services/MerchStoreService.ts

```diff
@@ -540,4 +540,14 @@ export default class MerchStoreService {
       await orderPickupEventRepository.deletePickupEvent(pickupEvent);
     });
   }
+
+  public async getCartItems(items: string[]): Promise<Map<string, PublicMerchItemOption>> {
+    return this.transactions.readOnly(async (txn) => {
+      const merchItemRepository = Repositories.merchStoreItemOption(txn);
+      const itemModels = await merchItemRepository.batchFindByUuid(items);
+      const publicItems = new Map<string, PublicMerchItemOption>();
+      itemModels.forEach(((v, k) => publicItems.set(k, v.getPublicMerchItemOption(false))));
+      return publicItems;
+    });
```

Comment:
The naming of the repositories is misleading here - we are using the `MerchItemOption` repository but are naming it as `merchItem` or `publicItems`. Remember, what the frontend wants is a list of options (to display the option chosen) and their item (to display the item name and image), so let's rename the variables accordingly to match that.

Also, the reason we are returning a `Map<string, PublicMerchItemOption>` is so that we are returning the data in the same order as the frontend sent it, so that the cart is displayed in the same order each time. If we're building our map based on `itemModels`, then the ordering of the map will be different than the ordering of uuids passed in `items`. So what we'd need to do is do `items.forEach((v, k)...)` instead of `itemModels.forEach((v, k)...)`, since then the map will be in the same order as the input. 

Additionally, we can lower the complexity of building the map by using `Array.prototype.map()` along with the `Map()` constructor:
```ts
return new Map(itemsOptionModels.map((option) => [option.uuid, option.getPublicMerchItemOption(false)]);
```
Let me know if anything's confusing with the above

Reply:
Sounds good, just pushed my implementation (itemOptionModels is actually a map but I think the current implementation should work).